### ISSUE TITLE
[XIF] Issue_req.id not updating and missing commit_valid

### ISSUE THREAD

**Issue by davidmallasen** [openhwgroup/cv32e40x#814]
### Component
Issues in the RTL

### Steps to Reproduce
Git hash: b66765acf650493901b6d3e7b44eb37bd2598f4a

Hello,

In the same setup as issue #800 , the `id` of the instruction through the `issue_req` interface is not updating (is always 0) and therefore the second time an instruction is offloaded, it gets stuck waiting for a `commit_valid` that never arrives.

I'm running this snippet of code, where it gets stuck after the issue of the second load instruction:
![Screenshot at 2023-03-08 15-53-59](https://user-images.githubusercontent.com/15904722/223768748-7380c21c-2138-4a08-8743-826816f78afe.png)

Firstly there is a random offloading that gets killed instantly:
![cv32e40x_1](https://user-images.githubusercontent.com/15904722/227554300-0f30b80c-d14e-4e08-8024-9b036af4c278.png)

Then the first load instruction offloads and has to wait for the `mem_ready`:
![cv32e40x_2](https://user-images.githubusercontent.com/15904722/227554450-5e1b94b8-1f2f-4df5-b9d4-2fc4172b8448.png)

Then the `mem_ready` arrives, the instruction completes, and the second load is issued. However, the `commit_valid` for this second instruction never arrives because (I presume) the id is the same as the previous ones:
![cv32e40x_3](https://user-images.githubusercontent.com/15904722/227554662-c9f88ba3-dd4a-4231-b585-80fc183baeb5.png)

I attach a VDC dump of the execution.
[waveform.vcd.zip](https://github.com/openhwgroup/cv32e40x/files/11063526/waveform.vcd.zip)

Thanks!
David

**Comment by Silabs-ArjanB** [comment]
Note that https://github.com/openhwgroup/cv32e40x/pull/825 was closed but its proposed fix has not been merged yet

**Comment by davidmallasen** [comment]
Correct, the closing of the PR was an unintended side-effect, sorry about that @Silabs-ArjanB
